docroot/modules/custom/sitenow_migrate/src/Plugin/migrate/source/BaseNodeSource.php

```diff
@@ -90,49 +90,43 @@ public function prepareRow(Row $row) {
   /**
    * Extract a plain text summary from a block of text.
    *
-   * @todo Use smart_trim for this.
-   *
-   * @param string $text
+   * @param string $output
    *   The text to convert to a trimmed plain text version.
    *
    * @return string
    *   The plain text string.
    */
-  protected function extractSummaryFromText(string $text) {
-    // Strip out any HTML, decode special characters and replace spaces.
-    $new_summary = Html::decodeEntities($text);
-    $new_summary = str_replace('&nbsp;', ' ', $new_summary);
-    // Also want to remove any excess whitespace on the left
-    // that might cause weird spacing for our summaries.
-    $new_summary = ltrim(strip_tags($new_summary));
+  protected function extractSummaryFromText(string $output) {
```

Comment:
It might be nice to be able to pass in the desired trim length. We could add an optional `$length` parameter to `getSummaryFromTextField` which defaults to 400 and then pass it to this method.

Reply:
Good call! Added it in to both getSummary and extractSummary.

Reply:
noice